### PR TITLE
Added new tests

### PR DESCRIPTION
**Pull request by agustinlg**
Added new tests for: 

- Code mirror error markers
- Raml editor adapter
- Dereference arrays
- Mocking service utils
- Save all

### CODE REVIEW

File: app/scripts/polyfills/array-polyfills.js [hunk 1]
```diff
@@ -9,3 +9,13 @@ if (!Array.prototype.find) {
     return undefined;
   };
 }
+
+if (!Array.prototype.includes) {
+  Array.prototype.includes = function (value) {
+    for (var i = 0; i < this.length; i++) {
```
Comment (llattes) [inline app/scripts/polyfills/array-polyfills.js]: Just a minor comment, please write this for loop as follows (length caching):
`for (var i = 0, len = this.length; i < len; i++) {`

File: test/spec/directives/raml-editor-save-file-button.js [hunk 2]
```diff
@@ -19,7 +28,16 @@ describe('ramlEditorSaveFileButton', function() {
...
         path: '/myFile.raml',
-        contents: 'some content'
+        contents: 'some content',
+        dirty: true
+      }
+    };
+    scope.homeDirectory = {
+      children: [scope.fileBrowser.selectedFile],
+      forEachChildDo: function(action) {
+        for (var i = 0; i < this.children.length; i++) {
```
Comment (llattes) [inline test/spec/directives/raml-editor-save-file-button.js]: Like before, move `this.children.length` to the initialization expression.

File: app/scripts/services/code-mirror-errors.js [hunk 3]
```diff
@@ -163,12 +163,11 @@
...
-        var editor = codeMirror.getEditor();
-        var annotations = groupByLine(annotationsNotSorted);
-
-        this.clearAnnotations();
+      service.displayAnnotations = function (annotationsNotSorted, editor) {
+        editor = editor || codeMirror.getEditor();
+        clearMarks(editor);
 
+        var annotations = groupByLine(annotationsNotSorted);
         for (var line = 0; line < annotations.length; ++line) {
```
Comment (llattes) [inline app/scripts/services/code-mirror-errors.js]: Same here, with `annotations.length`.

File: app/scripts/controllers/raml-editor-main.js [hunk 4]
```diff
@@ -4,7 +4,7 @@
   angular.module('ramlEditorApp')
     .constant('UPDATE_RESPONSIVENESS_INTERVAL', 800)
     .controller('ramlEditorMain', function (UPDATE_RESPONSIVENESS_INTERVAL, $scope, $rootScope, $timeout, $window,
-      safeApply, safeApplyWrapper, debounce, throttle, ramlParserAdapter, ramlRepository, codeMirror,
+      safeApply, safeApplyWrapper, debounce, ramlParserAdapter, ramlRepository, codeMirror,
```
Comment (llattes) [inline app/scripts/controllers/raml-editor-main.js]: `throttle` was declared but unused, right?

Comment (javok002) [inline app/scripts/controllers/raml-editor-main.js]: `throttle` was an old dependency that is not longer used.

**Pull request merged**